src/sources/exec/mod.rs

```diff
@@ -61,6 +62,13 @@ pub struct ExecConfig {
     #[configurable(metadata(docs::examples = "echo", docs::examples = "Hello World!"))]
     pub command: Vec<String>,
 
+    /// Custom environment variables to set or update when running the command.
```

Comment:
You'll want to use the `configurable` macro for both of these new options. For an example of another `HashMap` option see https://github.com/vectordotdev/vector/blob/ca7fa05ca98ac8ed097dc7a24b1652d62dbf283a/src/sinks/elasticsearch/config.rs#L147-L151

These then end up generating the cue files via `cargo vdev build component-docs` rather than them being hand edited as it seems you've done here.

Reply:
Ah yes, indeed I saw those configurable macros but wasn't sure how to use them.
I did try to use `cargo vdev build component-doc` but it gave a schema missing error and didn't know how to progress. By the way, I am using the `timberio/vector-dev` environment image, not a local toolchain.

I will implement the configurable macros and try to see what's wrong with `build component-docs` 👍.

Reply:
Ah, that's right, I forgot it's a two step process. There is a `make` target that handles both: `make generate-component-docs`.

Reply:
That worked! :)
Thanks for tips and tricks 👍

Reply:
Done in dd1e1d5fa932b0c6d28407de49fa6b3aafee5427
And docs automatically generted in a9c06485457463a4431a9995f7c20b8e5d18e934